src/components/ClockModule.js

```diff
@@ -0,0 +1,281 @@
+import React, { useCallback, useRef, useState } from 'react'
+import {
+  blockExplorerUrl,
+  Button,
+  ButtonBase,
+  GU,
+  IconCheck,
+  IconClock,
+  IconCross,
+  IconExternal,
+  textStyle,
+  Timer,
+  useTheme,
+} from '@aragon/ui'
+import { useWallet } from 'use-wallet'
+
+import HeaderModule from './Header/HeaderModule'
+import HeaderPopover from './Header/HeaderPopover'
+import useNetwork from '../hooks/useNetwork'
+import { useHeartbeat } from '../hooks/useCourtContracts'
+import { useCourtClock } from '../providers/CourtClock'
+import { useCourtConfig } from '../providers/CourtConfig'
+
+import { shortenAddress } from '../lib/web3-utils'
+import { formatDuration } from '../utils/date-utils'
+
+import logoSvg from '../assets/LogoAccent.svg'
+
+function ClockModule() {
+  const buttonRef = useRef()
+  const [opened, setOpened] = useState(false)
+
+  const theme = useTheme()
+  const wallet = useWallet()
+  const onHeartbeat = useHeartbeat()
+  const courtConfig = useCourtConfig()
+  const { type: networkType } = useNetwork()
+  const {
+    currentTermId,
+    currentTermEndDate,
+    isSynced,
+    neededTransitions,
+  } = useCourtClock()
+
+  const toggle = useCallback(() => setOpened(opened => !opened), [])
+
+  const handlePopoverClose = useCallback(() => {
+    setOpened(false)
+  }, [])
+
+  const handleOnClick = useCallback(async () => {
+    try {
+      const tx = await onHeartbeat(neededTransitions)
+      await tx.wait()
+    } catch (err) {
+      console.error(err)
+    }
```

Comment:
Note to self: remember to use [`addActivity()`](https://github.com/aragon/court-dashboard/blob/activity/src/components/Activity/ActivityProvider.js#L109) here, whenever #100 or this PR gets merged.

Reply:
Added in https://github.com/aragon/court-dashboard/pull/227/commits/f1b5e8d2f1ef093f28f39f709b132f82f3a1851a